src/components/input-field/input-field.tsx

```diff
@@ -268,7 +279,24 @@ export class InputField {
             return;
         }
 
-        return <p class="mdc-text-field-helper-text">{this.helperText}</p>;
+        return (
+            <p
+                class={`
+            ${
+                this.onValidateMinValueLength()
+                    ? 'mdc-text-field-helper-text--persistent mdc-theme-error'
+                    : 'mdc-text-field-helper-text'
+            }
+            
+        `}
+            >
+                {this.helperText}
+            </p>
+        );
+    }
+
+    onValidateMinValueLength() {
```

Comment:
It should not be the responsibility of this component to perform validation since it can never know when value is valid, thats why we have the `invalid` property. It's true that we can have some validation base on the `minlength`, `maxlength`, `min`, `max` and `required` properties, but that is handled by underlying `input` tag and can be checked with the either the `:invalid` CSS property (https://developer.mozilla.org/en-US/docs/Web/CSS/:invalid) or the Constraint Validation API (https://developer.mozilla.org/en-US/docs/Web/API/HTMLObjectElement/checkValidity)

The only other thing we can do to decide if the field is valid or not is to check the `invalid` property that is sent into the component. `invalid` should always take precedence